app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,33 @@
 # class for Posts entity
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = current_user
+    # @posts = @user.posts
+    @posts = @user.posts.includes(:comments, :likes)
+  end
 
-  def show; end
+  def show
+    @post = Post.find(params[:id])
+  end
+
+  def new
+    @post = Post.new
+    # render :new, locals: { post: post }
+  end
+
+  def create
+    @user = current_user
+    # @post = current_user.posts.create(title: params[:tile], text: params[:text])
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    # post.users = user
+    # post.users_id = user.id
+    @user.posts.push(@post)
```

Comment:
- [ ] <b>REQUIRED:</b> As the previous reviewer requested. to make sure that your codebase is always clean, remember to remove commented codes.